src/udev/fido_id/fido_id.c

```diff
@@ -67,7 +67,7 @@ static int run(int argc, char **argv) {
         if (!desc_path)
                 return log_oom();
 
-        fd = open(desc_path, O_RDONLY | O_NOFOLLOW | O_CLOEXEC);
+        fd = open(desc_path, O_RDONLY | O_NOFOLLOW | O_CLOEXEC | O_NONBLOCK | O_NOCTTY);
```

Comment:
hmm, O_NONBLOCK? the effect of that highly depends on backing inode. Sometimes it has an effect on open() behaviour and sometimes it does not. It always has an effect on read() and so on. 

On tty devices O_NONBLOCK disables waiting for carrier iirc, and on some block devices it disables waiting for medium insert iirc. Here we open a sysfs attribute, no idea what effect this has, if any.

Generally: I think it makes sense to open block devices with O_NONBLOCk and some other devices, but usually you want to turn it off right after, so it doesn't affect read(). 

Or in other words, I am not entirely sure it's really the right thing to do here.

(O_NOCTTY is always OK I figure, if we might end up opening the wrong inode at some place we might end up with a contorlling terminal, which we don't want. Unlike O_NOCTTY has very clear meaning across all inodes, so it should be riskless to specify)

Reply:
Interesting. Thank you for the explanation.